tests/suites/test_suite_pkwrite.function

```diff
@@ -124,3 +125,77 @@ void pk_write_key_check(char *key_file, int is_der)
     goto exit; /* make the compiler happy */
 }
 /* END_CASE */
+
+/* BEGIN_CASE */
+void pk_write_check_public_key_derivation(char *priv_key_file,
+                                          char *pub_key_file)
+{
+    mbedtls_pk_context priv_key, pub_key;
+    uint8_t derived_key_raw[PSA_EXPORT_PUBLIC_KEY_MAX_SIZE];
+    uint8_t *derived_key_start;
+    size_t derived_key_len = 0;
+    uint8_t pub_key_raw[PSA_EXPORT_PUBLIC_KEY_MAX_SIZE];
+    uint8_t *pub_key_start;
+    size_t pub_key_len = 0;
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    mbedtls_svc_key_id_t opaque_key_id = MBEDTLS_SVC_KEY_ID_INIT;
+#endif /* MBEDTLS_USE_PSA_CRYPTO */
+
+    mbedtls_pk_init(&priv_key);
+    mbedtls_pk_init(&pub_key);
+    USE_PSA_INIT();
+
+    memset(derived_key_raw, 0, sizeof(derived_key_raw));
+    memset(pub_key_raw, 0, sizeof(pub_key_raw));
+
+    TEST_EQUAL(mbedtls_pk_parse_keyfile(&priv_key, priv_key_file, NULL,
+                                        mbedtls_test_rnd_std_rand, NULL), 0);
+    TEST_EQUAL(mbedtls_pk_parse_public_keyfile(&pub_key, pub_key_file), 0);
+
+    /* mbedtls_pk_write_pubkey() writes data backward in the provided buffer,
+     * i.e. derived_key_raw, so we place derived_key_start at the end of it
+     * and it will be updated accordingly on return.
+     * The same holds for pub_key_raw and pub_key_start below.*/
+    derived_key_start = derived_key_raw + sizeof(derived_key_raw);
+    TEST_LE_U(1, mbedtls_pk_write_pubkey(&derived_key_start,
+                                         derived_key_raw, &priv_key));
+    derived_key_len = sizeof(derived_key_raw) -
+                      (derived_key_start - derived_key_raw);
+
+
+    pub_key_start = pub_key_raw + sizeof(pub_key_raw);
+    TEST_LE_U(1, mbedtls_pk_write_pubkey(&pub_key_start,
```

Comment:
I think this strategy works, but to me it seems a bit more convoluted than needed: we part the public key into a PK context then write it out. I think we could just directly read the file into a buffer using `mbedtls_pk_load_file()`¹, no need for a second PK context. IMO having only one PK context would make the code a bit clearer too, and more similar to the other test functions.

¹ No need for `fix_new_lines()` since this is DER.

Wdyt?